rollbar_common/lib/src/data/breadcrumb_record.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:meta/meta.dart';
+import '../../rollbar_common.dart'
+    show
+        Persistable,
+        PersistableFor,
+        SerializableFor,
+        UUID,
+        uuidGen,
+        Datatype,
+        JsonMap;
+
+@sealed
+@immutable
+class BreadcrumbRecord implements Persistable<UUID> {
```

Comment:
Yes, I 1000% agree, and that's how I planned it.

However, there's still a bunch of limitations on Dart that forced me to move `BreadcrumRecord` and `PayloadRecord` to `rollbar_common`.

The Serializable and Persistable interfaces should be very simple. Ie. Serializable describes the class of types that can be instantiated with a json payload, and that can give you a json representation of themselves, right? So that's two functions, a constructor `fromJson` and a method `toJson`.

Trouble is, Dart doesn't support abstract static members nor constructors. 🙄 
* https://github.com/dart-lang/language/issues/356

What you see in `Persistable.dart` and `Serializable.dart` is a hacky workaround to express this idea of Persistable and Serializable objects, the downside is that both Persistable and Serializable _need to know_ the objects that implement them. I know. Utterly horrible.

So, the goal is to change this whenever Dart gets that feature, and leave this like this because playing with these abstract interfaces instead of the objects themselves is much simpler and forces me to think generically or abstractly.

At the very least, it's not the actual Breadcrumb/Payload that's in common, but the record. 🤷‍♂️

Reply:
There are plenty of limitations of the Dart type system that made the implementation harder than it should be:

- https://github.com/dart-lang/language/issues/349
- https://github.com/dart-lang/language/issues/356
- https://github.com/dart-lang/language/issues/1152
- https://github.com/dart-lang/language/issues/359

The good thing is that the generic parts are generic enough that this can be used as intended.

Reply:
Yes, I agree you had enough external reasons to place it in the common package. That is why I made a comment "to keep in mind" for future refactoring. It would be a nice improvement if possible. Maybe by defining some generic interfaces in the common package and then moving more concrete implementation to the dart package. In either case, it is not a show-stopper by any means.